### ISSUE TITLE
Fonts break with -pin_fonts_to_cache=false -draw_blurs=false

### ISSUE THREAD

**Issue by divVerent** [divVerent/aaaaxy#68]
That option should only reduce performance by not calling text.CacheGlyphs(), but it's actually breaking all font rendering.

This issue was apparently introduced in 91a0526fd409de17baa113f1db11cf35045816dc which updated Ebitengine:

```
-       github.com/hajimehoshi/ebiten/v2 v2.1.4
+       github.com/hajimehoshi/ebiten/v2 v2.2.0-alpha.11.0.20210817032719-12350e2fee5d
```

This issue becomes more relevant now as glyph precaching no longer works in Ebitengine 2.5 due to better font hinting support.

**Comment by divVerent** [comment]
Suggest starting in https://github.com/divVerent/aaaaxy/commit/91a0526fd409de17baa113f1db11cf35045816dc and bisecting the Ebitengine versions to learn more.

**Comment by divVerent** [comment]
Also, as for glyph precaching, need to see if we need to "make it work anyway", i.e. break hinting so we get caching again.

**Comment by divVerent** [comment]
Problematic Ebitengine commit:

```
commit 2a8e8d0eb66ec909486b75c4717ff00d33c1b2d8
Author: Hajime Hoshi <hajimehoshi@gmail.com>
Date:   Thu Jun 24 12:49:36 2021 +0900

    internal/graphicsdriver/opengl: Stop using PBO
    
    PBO was introduced to improve the performance at ReplacePixels,
    but we found that PBO can degrades the performance. Also, now
    multiple glTexImage2Ds are called successively like a batch, so
    the situation is now different from that time when PBO was
    introduced.
    
    Let's remove PBO usages and wait and see.
    
    Closes #1678
```

**Comment by divVerent** [comment]
non-PBO code was:

```
i.graphics.context.texSubImage2D(i.textureNative, args)
```

which does

```
        c.bindTexture(t)
        for _, a := range args {
                gl.TexSubImage2D(gl.TEXTURE_2D, 0, int32(a.X), int32(a.Y), int32(a.Width), int32(a.Height), gl.RGBA, gl.UNSIGNED_BYTE, gl.Ptr(a.Pixels))
        }
```

Nothing really suspicious.

**Comment by divVerent** [comment]
What if the texture t got reallocated in the meantime? Stupid idea, but maybe?

**Comment by divVerent** [comment]
Created an apitrace that shows the problem.

[aaaaxy.25839.trim.trace.zip](https://github.com/divVerent/aaaaxy/files/10365070/aaaaxy.25839.trim.trace.zip)

Something wrong with the texture indeed:

![20230106-192306](https://user-images.githubusercontent.com/251568/211121672-b7474367-13dc-4f6d-89c4-c0cc29cfb72d.png)

**Comment by divVerent** [comment]
At time of call, GL_TEXTURE0 is a texture of my game (the visibility mask). This is odd.

**Comment by divVerent** [comment]
Suspicion: when calling glActiveTexture, glBindTexture state changes, and necessary texture switches are not emitted...?

**Comment by divVerent** [comment]
Yeah, have a patch for Ebitengine that fixes this.

**Comment by divVerent** [comment]
Fixed in upstream main - closing here now; will be fixed once switching over to Ebitengine 2.5.
